source/shared/cpp/ObjectModel/Util.cpp

```diff
@@ -76,43 +76,69 @@ std::string ValidateColor(const std::string& backgroundColor,
 }
 
 void ValidateUserInputForDimensionWithUnit(const std::string &unit, const std::string &requestedDimension,
-    int &parsedDimension)
+    int &parsedDimension, std::vector<std::shared_ptr<AdaptiveCardParseWarning>>& warnings)
 {
-    if (requestedDimension.empty())
-    {
-        parsedDimension = 0;
-    }
-    else
+    const std::string warningMessage = "expected input arugment to be specified as \\d+(\\.\\d+)?px with no spaces, but received ";
+    parsedDimension = 0;
+    std::string stringPattern = "^([1-9]+\\d*)(\\.\\d+)?";
+    stringPattern += ("(" + unit + ")$");
+    std::regex pattern(stringPattern);
+    std::smatch matches;
+
+    if(std::regex_search(requestedDimension, matches, pattern))
     {
-        const std::size_t foundIndex = requestedDimension.find(unit);
-        if (std::string::npos == foundIndex || requestedDimension.size() != foundIndex + unit.size())
+        try
         {
-            throw AdaptiveCardParseException(ErrorStatusCode::InvalidPropertyValue,
-                "unit is either missing or inproper form: " + requestedDimension);
+            // stoi will get integral value upto non digit char indexed by idx
+            parsedDimension = std::stoi(matches[0]);
         }
-        try
+        catch (const std::invalid_argument &)
         {
-            int parsedVal = std::stoi(requestedDimension.substr(0, foundIndex));
-            if (parsedVal < 0)
-            {
-                throw AdaptiveCardParseException(ErrorStatusCode::InvalidPropertyValue,
-                    "unsigned integer is accepted but received : " + requestedDimension);
-            }
-            parsedDimension = parsedVal;
+            warnings.emplace_back(std::make_shared<AdaptiveCardParseWarning>(
+                    AdaptiveSharedNamespace::WarningStatusCode::InvalidDimensionSpecified,
+                    warningMessage + requestedDimension)); 
         }
-        catch (const std::invalid_argument &e)
+        catch (const std::out_of_range &)
         {
-            (void)e;
-            throw AdaptiveCardParseException(ErrorStatusCode::InvalidPropertyValue,
-                "unsigned integer is accepted but received : " + requestedDimension);
+            warnings.emplace_back(std::make_shared<AdaptiveCardParseWarning>(
+                    AdaptiveSharedNamespace::WarningStatusCode::InvalidDimensionSpecified,
+                    "out of range: " + requestedDimension));
         }
-        catch (const std::out_of_range &e)
+    }
+    else 
+    {
+        warnings.emplace_back(std::make_shared<AdaptiveCardParseWarning>(
+                AdaptiveSharedNamespace::WarningStatusCode::InvalidDimensionSpecified,
+                warningMessage + requestedDimension)); 
+    }
+}
+
+bool ShouldParseForExplicitDimension(const std::string &input)
```

Comment:
can't the logic in this function be replaced with a regex as well?

Reply:
it would have been great fit if we were looking for specific pattern, but we are checking if the string contains specific char combinations. we are doing this, so we can produce relavant warning messages. if user misspelled "auto", we don't want to say "it doesn't contain px". 

---
In reply to: [217520663](https://github.com/Microsoft/AdaptiveCards/pull/1872#discussion_r217520663) [](ancestors = 217520663)